### PR TITLE
[#166734445] Remove awilix

### PR DESCRIPTION
**Pull request by BurnedMarshal**
This PR has all the changes from other PR #462 #470 #476 rebased with master

### CODE REVIEW

**Comment by digitalcitizenship** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Please include a description of your PR changes.
  </td>
    </tr>
  </tbody>
</table>



## Affected stories

  * 🌟 [#166734445](https://www.pivotaltracker.com/story/show/166734445): Rimuovere awilix dal codice

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/app.ts [hunk 1]
```diff
@@ -45,29 +38,43 @@ import {
...
+import MessagesService from "./services/messagesService";
+import PagoPAProxyService from "./services/pagoPAProxyService";
+import ProfileService from "./services/profileService";
+import RedisSessionStorage from "./services/redisSessionStorage";
+import RedisUserMetadataStorage from "./services/redisUserMetadataStorage";
+import TokenService from "./services/tokenService";
+
 const defaultModule = {
-  loadSpidStrategy,
+  currentSpidStrategy: undefined as SpidStrategy | undefined,
```
Comment (gunzip) [inline src/app.ts]: is this a logical or ? (it returns `0`)

Comment (BurnedMarshal) [reply src/app.ts]: maybe this is the correct version
```suggestion
  currentSpidStrategy: undefined as (SpidStrategy | undefined),
```

Comment (gunzip) [reply src/app.ts]: sorry that was correct (but not very clear ! : )

Comment (BurnedMarshal) [reply src/app.ts]: Ok, I'll resume the previous version ;)

File: .circleci/config.yml [hunk 2]
```diff
@@ -98,6 +98,8 @@ jobs:
           command: yarn test
           environment:
             PRE_SHARED_KEY: "12345"
+            API_KEY: "fake_api_key_here"
+            API_URL: "https://agid-apim-prod.azure-api.net/api/v1"
```
[the remark this answers is not in the document]
Comment (BurnedMarshal) [reply .circleci/config.yml]: Fixed in 3a511c4

File: .circleci/config.yml [hunk 3]
```diff
@@ -98,6 +98,8 @@ jobs:
           command: yarn test
           environment:
             PRE_SHARED_KEY: "12345"
+            API_KEY: "fake_api_key_here"
+            API_URL: "https://api.gateway.com/api/v1"
```
Comment (gunzip) [inline .circleci/config.yml]:
```suggestion
            API_URL: "https://api.gateway.example.com/api/v1"
```

**Comment by gunzip** [comment]
:rocket:

**Pull request merged**